### ISSUE TITLE
METS.xml cannot be processed properly with `ocrd/all:maximum` (ID: `2d9b243e9315`)

### ISSUE THREAD

**Issue by mweidling** [OCR-D/core#1112]
I tried to process to following workspace (as an example for the GTs used in Quiver) using Docker images of `ocrd/all`:

[alberti.zip](https://github.com/OCR-D/ocrd_all/files/12878563/alberti.zip)

When I try to process this workspace with an older version of `ocrd/all`, everything works fine, e.g.

`docker run --rm -v $PWD:/data ocrd/all:20-23-06-28 ocrd-olena-binarize -I OCR-D-IMG -O OCR-D-BIN -P impl sauvola`

yields the expected binarization.

Doing the same with `ocrd/all:maximum` (image ID `2d9b243e9315`) (`docker run --rm -v $PWD:/data ocrd/all:maximum ocrd-olena-binarize -I OCR-D-IMG -O OCR-D-BIN -P impl sauvola`) yield the following error:

```
Traceback (most recent call last):
  File "/usr/local/bin/ocrd", line 33, in <module>
    sys.exit(load_entry_point('ocrd', 'console_scripts', 'ocrd')())
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/build/core/ocrd/ocrd/cli/validate.py", line 113, in validate_process
    _inform_of_result(validate_tasks([ProcessorTask.parse(t) for t in tasks],
  File "/build/core/ocrd/ocrd/task_sequence.py", line 115, in validate_tasks
    raise Exception("Invalid task sequence input/output file groups: %s" % report.errors)
Exception: Invalid task sequence input/output file groups: ["Input fileGrp[@USE='OCR-D-IMG'] not in METS!"]
```

Can you reproduce this behavior? `fileGrp[@USE='OCR-D-IMG']` is definitely part of the workspace's METS file, otherwise the first process would not have been successful.

**Comment by mweidling** [comment]
This issue might have been introduced in https://github.com/OCR-D/core/releases/tag/v2.54.0, no?

**Comment by kba** [comment]
I can confirm the issue, getting the same error with the docker call. Getting a slightly more helpful response when running v1.3.0 of ocrd_olena natively with current core:

```
14:20:18.499 ERROR ocrd.ocrd-olena-binarize - input file ID=OCR-D-IMG_0008 (pageId=phys_0008 MIME=image/tiff) is not on disk   
14:20:19.744 ERROR ocrd.ocrd-olena-binarize - input file ID=OCR-D-IMG_0009 (pageId=phys_0009 MIME=image/tiff) is not on disk   
14:20:21.027 ERROR ocrd.ocrd-olena-binarize - input file ID=OCR-D-IMG_0007 (pageId=phys_0007 MIME=image/tiff) is not on disk   
ocrd-olena-binarize -I OCR-D-IMG -O OCR-D-BIN -P impl sauvola  23.11s user 7.10s system 148% cpu 20.356 total                  
```

And with the latest v1.4.0 the error goes away, at least natively:


```
14:21:20.886 INFO ocrd.ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0008 (phys_0008)
14:21:24.960 INFO ocrd.ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0009 (phys_0009)
14:21:29.054 INFO ocrd.ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0007 (phys_0007)
14:21:31.970 INFO ocrd.cli.workspace.bulk-add - [   1/6] OCR-D-BIN image/png phys_0008 OCR-D-BIN_0008-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0008-BIN_sauvola.png
14:21:31.972 INFO ocrd.cli.workspace.bulk-add - [   2/6] OCR-D-BIN application/vnd.prima.page+xml phys_0008 OCR-D-BIN_0008 OCR-D-BIN/OCR-D-BIN_0008.xml
14:21:31.974 INFO ocrd.cli.workspace.bulk-add - [   3/6] OCR-D-BIN image/png phys_0009 OCR-D-BIN_0009-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0009-BIN_sauvola.png
14:21:31.975 INFO ocrd.cli.workspace.bulk-add - [   4/6] OCR-D-BIN application/vnd.prima.page+xml phys_0009 OCR-D-BIN_0009 OCR-D-BIN/OCR-D-BIN_0009.xml
14:21:31.976 INFO ocrd.cli.workspace.bulk-add - [   5/6] OCR-D-BIN image/png phys_0007 OCR-D-BIN_0007-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0007-BIN_sauvola.png
14:21:31.977 INFO ocrd.cli.workspace.bulk-add - [   6/6] OCR-D-BIN application/vnd.prima.page+xml phys_0007 OCR-D-BIN_0007 OCR-D-BIN/OCR-D-BIN_0007.xml
```

Now building new ocrd_all docker and verify that it's fixed there as well.

TLDR: Error will be solved with new ocrd_all.

**Comment by mweidling** [comment]
Thanks for tending to this! I'll re-check with the new ocrd_all docker.

**Comment by kba** [comment]
OK, this is getting bizarre.

The `mets.xml` contains `mets:fileGrp` but they are not within `mets:fileSec`. 

> If it helps: I realized that the METS structure that is created with `docker run --rm -v $PWD:/data ocrd/all:maximum ocrd workspace init` plus adding the images as [described on our website](https://ocr-d.de/en/user_guide#non-existing-mets) differs from the METS structure I have in my workspace above. Processes using this newly created METS file work like a charm.

Are you sure that there was no additional processing of that workspace?

At first I thought it was a bug in `ocrd bag zip` but in fact `ocrd zip bag` **fixes** this issue, adding the missing `mets:fileSec` wrapper.

The other bizarre behavior is that all log output is missing in docker. But the processing works fine now, as expected in the latest ocrd_all release v2023-10-17.

Final bizarre behavior is that in the native installation

```
ocrd workspace find -k fileGrp -k local_filename
OCR-D-IMG       OCR-D-IMG/OCR-D-IMG_0008.tif
OCR-D-IMG       OCR-D-IMG/OCR-D-IMG_0009.tif
OCR-D-IMG       OCR-D-IMG/OCR-D-IMG_0007.tif
OCR-D-GT-SEG-PAGE       OCR-D-GT-SEG-PAGE/OCR-D-GT-SEG-PAGE_0008.xml
OCR-D-GT-SEG-PAGE       OCR-D-GT-SEG-PAGE/OCR-D-GT-SEG-PAGE_0009.xml
OCR-D-GT-SEG-PAGE       OCR-D-GT-SEG-PAGE/OCR-D-GT-SEG-PAGE_0007.xml
OCR-D-GT-SEG-BLOCK      OCR-D-GT-SEG-BLOCK/OCR-D-GT-SEG-BLOCK_0008.xml
OCR-D-GT-SEG-BLOCK      OCR-D-GT-SEG-BLOCK/OCR-D-GT-SEG-BLOCK_0009.xml
OCR-D-GT-SEG-BLOCK      OCR-D-GT-SEG-BLOCK/OCR-D-GT-SEG-BLOCK_0007.xml
OCR-D-GT-SEG-LINE       OCR-D-GT-SEG-LINE/OCR-D-GT-SEG-LINE_0008.xml
OCR-D-GT-SEG-LINE       OCR-D-GT-SEG-LINE/OCR-D-GT-SEG-LINE_0009.xml
OCR-D-GT-SEG-LINE       OCR-D-GT-SEG-LINE/OCR-D-GT-SEG-LINE_0007.xml
```

But in docker:

```
docker run --rm -it -v $PWD:/data ocrd/all ocrd workspace find -k fileGrp -k local_filename

```

I.e. no output at all without the `fileSec`. I.e. there must be something different in the XML handling native/docker.

**Comment by mweidling** [comment]
> Are you sure that there was no additional processing of that workspace?

As for the newly created workspace: Yes, I ran

* `docker run --rm -u $(id -u) -v $PWD:/data -- ocrd/all:maximum ocrd workspace [-d path/to/your/workspace] init`
* `docker run --rm -u $(id -u) -v $PWD:/data -- ocrd/all:maximum ocrd workspace [-d path/to/your/workspace] set-id 'unique ID'`
* … and the bash script for adding images to the workspace

However, I cannot say how exactly the mets file in the problematic workspace has been created. This has been done by @tboenig several months ago, and I did not change it.

I re-checked the original mets.xml (the one without `mets:fileSec`) against the METS schema and it's not valid. I see nothing in the OCR-D METS specs that might hint at being allowed to leave `mets:fileSec` out. **So I inserted the missing `mets:fileSec`and observe the following:**

__1. Using the former release:__

`docker run --rm -v $PWD:/data ocrd/all:2023-06-28 ocrd-olena-binarize -I OCR-D-IMG -O OCR-D-BIN -P impl sauvola`:

```
07:35:35.949 INFO ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0008 (phys_0008)
07:35:48.558 INFO ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0009 (phys_0009)
07:35:59.269 INFO ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0007 (phys_0007)
07:36:07.167 INFO ocrd.cli.workspace.bulk-add - [   1/6] OCR-D-BIN image/png phys_0008 OCR-D-BIN_0008-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0008-BIN_sauvola.png
07:36:07.168 INFO ocrd.cli.workspace.bulk-add - [   2/6] OCR-D-BIN application/vnd.prima.page+xml phys_0008 OCR-D-BIN_0008 OCR-D-BIN/OCR-D-BIN_0008.xml
07:36:07.169 INFO ocrd.cli.workspace.bulk-add - [   3/6] OCR-D-BIN image/png phys_0009 OCR-D-BIN_0009-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0009-BIN_sauvola.png
07:36:07.169 INFO ocrd.cli.workspace.bulk-add - [   4/6] OCR-D-BIN application/vnd.prima.page+xml phys_0009 OCR-D-BIN_0009 OCR-D-BIN/OCR-D-BIN_0009.xml
07:36:07.170 INFO ocrd.cli.workspace.bulk-add - [   5/6] OCR-D-BIN image/png phys_0007 OCR-D-BIN_0007-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0007-BIN_sauvola.png
07:36:07.170 INFO ocrd.cli.workspace.bulk-add - [   6/6] OCR-D-BIN application/vnd.prima.page+xml phys_0007 OCR-D-BIN_0007 OCR-D-BIN/OCR-D-BIN_0007.xml
```

**2. Using the latest release (2023-10-17):**

`docker run --rm -v $PWD:/data ocrd/all:maximum ocrd-olena-binarize -I OCR-D-IMG -O OCR-D-BIN -P impl sauvola`:

No output, but `olena` has written a `<mets:fileGrp USE="OCR-D-BIN">` into the mets file.

**3. And finally the problematic image with ID `2d9b243e9315`:**

`docker tag 2d9b243e9315 ocrd/all:test` (seems like the problematic image has been dangling?)

`docker run --rm -v $PWD:/data ocrd/all:test ocrd-olena-binarize -I OCR-D-IMG -O OCR-D-BIN -P impl sauvola`:

```
07:50:27.785 INFO ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0008 (phys_0008)
07:50:39.953 INFO ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0009 (phys_0009)
07:50:52.485 INFO ocrd-olena-binarize - processing image/tiff input file OCR-D-IMG_0007 (phys_0007)
07:51:02.075 INFO ocrd.cli.workspace.bulk-add - [   1/6] OCR-D-BIN image/png phys_0008 OCR-D-BIN_0008-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0008-BIN_sauvola.png
07:51:02.076 INFO ocrd.cli.workspace.bulk-add - [   2/6] OCR-D-BIN application/vnd.prima.page+xml phys_0008 OCR-D-BIN_0008 OCR-D-BIN/OCR-D-BIN_0008.xml
07:51:02.077 INFO ocrd.cli.workspace.bulk-add - [   3/6] OCR-D-BIN image/png phys_0009 OCR-D-BIN_0009-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0009-BIN_sauvola.png
07:51:02.077 INFO ocrd.cli.workspace.bulk-add - [   4/6] OCR-D-BIN application/vnd.prima.page+xml phys_0009 OCR-D-BIN_0009 OCR-D-BIN/OCR-D-BIN_0009.xml
07:51:02.077 INFO ocrd.cli.workspace.bulk-add - [   5/6] OCR-D-BIN image/png phys_0007 OCR-D-BIN_0007-BIN_sauvola OCR-D-BIN/OCR-D-BIN_0007-BIN_sauvola.png
07:51:02.078 INFO ocrd.cli.workspace.bulk-add - [   6/6] OCR-D-BIN application/vnd.prima.page+xml phys_0007 OCR-D-BIN_0007 OCR-D-BIN/OCR-D-BIN_0007.xml
```

TLDR; Everything works fine as long as the METS files are valid. Something in the more recent releases has changed so that the process is stricter when it comes to METS files being valid.

What's the consequence? IMO we could very well expect people (including ourselves ;) ) expect to use valid METS files. What do you think?

**Comment by mweidling** [comment]
> OK, this is getting bizarre.

I second that. https://github.com/tboenig/16_ant_complex/blob/main/data/alberti_pictura_1540/mets.xml#L194 has the `mets:fileSec`, but when I retrieve the latest release from this GT repository, said element is missing. But that's an issue for another repo.
